src/Functions/FunctionsHashing.h

```diff
@@ -274,6 +274,54 @@ struct SipHash64Impl
     static constexpr bool use_int_hash_for_pods = false;
 };
 
+struct SipHash64KeyedImpl
+{
+    static constexpr auto name = "sipHash64Keyed";
+    using ReturnType = UInt64;
+    struct KeyType
+    {
+        UInt64 key0 = 0;
+        UInt64 key1 = 0;
+    };
+
+    static KeyType parseKey(const ColumnWithTypeAndName & key)
+    {
+        KeyType ret;
+
+        const auto * tuple = checkAndGetColumn<ColumnTuple>(key.column.get());
+        if (!tuple)
+            throw Exception(ErrorCodes::NOT_IMPLEMENTED, "key must be a tuple");
+
+        if (tuple->tupleSize() != 2)
+            throw Exception(ErrorCodes::NOT_IMPLEMENTED, "wrong tuple size: key must be a tuple of 2 UInt64");
```

Comment:
Totally minor, but the check at l. 295 only checks that the input is a 2-valued tuple and not the actual data types (l. 296). Maybe rephrase: "key argument must be a tuple with two elements"

Reply:
You can do it how you like, any alternative is fine with me.